android/core/src/main/java/com/stadiamaps/ferrostar/core/NavigationViewModel.kt

```diff
@@ -104,14 +106,17 @@ interface NavigationViewModel {
  */
 class DefaultNavigationViewModel(
     private val ferrostarCore: FerrostarCore,
+    private val annotationPublisher: AnnotationPublisher<*> = NoOpAnnotationPublisher()
 ) : ViewModel(), NavigationViewModel {
 
   private val muteState: StateFlow<Boolean?> =
       ferrostarCore.spokenInstructionObserver?.muteState ?: MutableStateFlow(null)
 
   override val uiState =
       combine(ferrostarCore.state, muteState) { a, b -> a to b }
-          .map { (coreState, muteState) ->
+          .map { (coreState, muteState) -> annotationPublisher.map(coreState) to muteState }
+          .map { (stateWrapper, muteState) ->
```

Comment:
@ianthetechie might recommend `Triple`.

Reply:
Ahh, I see after pulling it up in Android Studio why you chose this method now... we can discuss briefly on the call, but I'm OK with this 👍